odoo/__init__.py

```diff
@@ -23,8 +23,32 @@
 # ----------------------------------------------------------
 # Shortcuts
 # ----------------------------------------------------------
-# The hard-coded super-user id (a.k.a. administrator, or root user).
-SUPERUSER_ID = 1
+# store file names from which the warning was raised
+# used to stop raising warnings after X times
+_GLOBAL_VARIABLE_WARNING = set()
+
+
+def __getattr__(name: str):
+    """Get variables from the odoo module and show warnings"""
+    match name:
+        case 'SUPERUSER_ID':
+            module_name = 'odoo.api'
+        case _:
+            raise AttributeError(f"Module {__name__!r} has not attribute {name!r}.")
+
+    # import the specific package (not just odoo)
+    module = __import__(module_name, fromlist=module_name.split('.'))
+
+    import inspect  # noqa: PLC0415
+    import warnings  # noqa: PLC0415
+
+    frame = inspect.currentframe().f_back
+    _GLOBAL_VARIABLE_WARNING.add(frame.f_code.co_filename)
+
+    if len(_GLOBAL_VARIABLE_WARNING) < 10:
```

Comment:
Let's not fill that set after 10 hits
```suggestion
    if len(_GLOBAL_VARIABLE_WARNING) < 10:
        _GLOBAL_VARIABLE_WARNING.add(frame.f_code.co_filename)
```